core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -217,18 +211,20 @@ public boolean hasSpaceForAnotherRecord() {
   }
 
   public void expandPointerArray(LongArray newArray) {
-    if (newArray.size() < array.size()) {
-      // checkstyle.off: RegexpSinglelineJava
-      throw new SparkOutOfMemoryError("Not enough memory to grow pointer array");
-      // checkstyle.on: RegexpSinglelineJava
+    if (array != null) {
+      if (newArray.size() < array.size()) {
+        // checkstyle.off: RegexpSinglelineJava
```

Comment:
Any idea why this is needed?

Reply:
No clue. The lines with the comments are newer than the line between them, so I'm sure they're good for something...